spec/models/info_request_spec.rb

```diff
@@ -94,37 +94,38 @@
 
   describe '#law_used' do
 
-    it 'sets the default law used' do
+    it 'defaults law used to foi' do
       expect(InfoRequest.new.law_used).to eq('foi')
     end
 
-    it 'sets the default law used to the legislation key' do
-      legislation = FactoryBot.build(:legislation, key: 'eir')
-      allow_any_instance_of(InfoRequest).to receive(:legislation).
-        and_return(legislation)
-      expect(InfoRequest.new(law_used: nil).law_used).to eq('eir')
-    end
-
-    it 'does not try to overwrite the existing law used' do
-      legislation = FactoryBot.build(:legislation, key: 'eir')
-      allow_any_instance_of(InfoRequest).to receive(:legislation).
-        and_return(legislation)
-      expect(InfoRequest.new(law_used: 'foi').law_used).to eq('foi')
-    end
-
     context 'with public body' do
 
+      let(:foi) { FactoryBot.build(:public_body) }
       let(:eir) { FactoryBot.build(:public_body) }
 
       before do
+        allow(foi).to receive(:legislation).and_return(
+          FactoryBot.build(:legislation, key: 'foi')
+        )
+
         allow(eir).to receive(:legislation).and_return(
           FactoryBot.build(:legislation, key: 'eir')
         )
       end
 
       it 'sets law used to the public body legislation on validataion' do
         request = FactoryBot.build(:info_request, public_body: eir)
-        expect(request.law_used).to eq('eir')
+
```

Comment:
1. The initialisation spec won't pass as public body is `nil` within in `after_initialize` block
2. Yes - of course we can use the `eir_only` trait

Reply:
3. Instead of squashing to remove the failing spec - I think its useful to show the actual issue - I will mark as `xit` so the suite is still green for that commit.